packages/terra-date-input/src/DateInput.jsx

```diff
@@ -442,6 +468,38 @@ class DateInput extends React.Component {
     }
   }
 
+  /**
+   * Sets state to new value dependent on existing date
+   * @param {Object} event Event object generated from the event delegation.
+   * @param {Number} addDays Adds days to current date or today's date.
+   */
+  setHotKeyDate(event, addDays) {
+    let { year, month, day } = this.state;
+    year = Number(year);
+    month = Number(month) - 1;
+    day = Number(day);
+    const dateFromInput = new Date(year, month, day);
+    const validYear = (year < DateInputUtil.MaxYearValue && year > DateInputUtil.MinYearValue && dateFromInput.getFullYear() === year);
+    const validMonth = (dateFromInput.getMonth() === month);
+    const validDay = (day > 0 && day <= 31 && dateFromInput.getDate() === day);
```

Comment:
You don't need to check if the day is within 1 and 31 anymore, because `dateFromInput.getDate() === day` will be false if day falls outside of the valid days for that month. Same if day is "cat", -1, etc. 

Try boundary testing on my codepen and you'll see these cases are correctly invalid dates w/o those extra checks:
* day= '0'
* day= '-1'
* day= 'cat'
* day=(one day over the last day of the month)

Reply:
Removed here: https://github.com/cerner/terra-framework/pull/1483/commits/2a6582f14d8b118be226db1e1816aa5418005e26#diff-ea35ec190c52db171ef5f774b071e80db0b038abff0e19580b310959fc582329R484